core/src/main/scala/org/apache/spark/resource/ExecutorResourceRequest.scala

```diff
@@ -63,15 +56,21 @@ private[spark] class ExecutorResourceRequest(
     val discoveryScript: String = "",
```

Comment:
I know this is not part of this PR, but shouldn't discoveryScript not be Option[String] instead of String with default "" ?

Reply:
I specifically made it "" instead of Option for the Java API compatibility rather than creating separate api's.  Did you have other suggestions?

Reply:
That is a good point. What about using Optional instead then ?

Reply:
yes should be able to. I wasn't sure if the Scala side would get annoyed with having to use java object as parameter, or more so that spark had any api's that did that. Maybe what I'll do is change ExecutorResourceRequest which I don't expect the user to wield directly as much to Optional and then provide 2 different resource() api's here,  one that take java Optional and one that takes the scala Option and convert to Optional before creating the ExecutorResourceRequest, just as a convenience thing for scala users.  thoughts?

Reply:
actually I looked at Optional and one issue is its not serializable. so that would make things a bit of a hassle.